source/Applications/Advanced/TransformViaArucoMarker/TransformViaArucoMarker.cpp

```diff
@@ -0,0 +1,315 @@
+/*
+This example shows how to transform a point cloud using the pose of an aruco marker.
+*/
+
+#include <Zivid/Experimental/Calibration.h>
+#include <Zivid/Visualization/Visualizer.h>
+#include <Zivid/Zivid.h>
+
+#include <algorithm>
+#include <opencv2/core/core.hpp>
+#include <opencv2/highgui/highgui.hpp>
+#include <opencv2/imgproc/imgproc.hpp>
+
+#include <aruco_nano.h>
+#include <cmath>
+#include <iostream>
+
+namespace
+{
+    struct Line2d
+    {
+        double slope;
+        double intercept;
+    };
+
+    int clamp(const int n, const int lower, const int upper)
+    {
+        return std::max(lower, std::min(n, upper));
+    }
+
+    // Fit a line y=a*x + b to 2 points
+    Line2d fitLine(const cv::Point2d &point1, const cv::Point2d &point2)
+    {
+        const auto slope{ (point2.y - point1.y) / (point2.x - point1.x) };
+        const auto intercept{ point1.y - slope * point1.x };
+        return { slope, intercept };
+    };
+
+    cv::Mat pointCloudToCvBGR(const Zivid::PointCloud &pointCloud)
+    {
+        auto rgb = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        pointCloud.copyData(
+            reinterpret_cast<Zivid::ColorRGBA *>(rgb.data)); // NOLINT(cppcoreguidelines-pro-type-reinterpret-cast)
+        auto bgr = cv::Mat(pointCloud.height(), pointCloud.width(), CV_8UC4); // NOLINT(hicpp-signed-bitwise)
+        cv::cvtColor(rgb, bgr, cv::COLOR_RGBA2BGR);
+
+        return bgr;
+    }
+
+    std::vector<cv::Point2f> estimate2DMarkerCenters(const aruconano::Marker &markers)
```

Comment:
Why is 'markers' plural? Isn't it a single marker?

Reply:
This becomes difficult to understand at lines 61-64 since you extract a corner from marker**s**.

Reply:
Or can this actually hold multiple markers and then the corners are stored like [marker1corner1, marker1corner2, marker1corner3, marker1corner4, marker2corner1, marker2corner2,...]?

Reply:
Actually the code can detect multiple markers and for each marker find the corner and pose. But, from our discussion of this and the ROI sample, i can edit it to only detect a single marker.